scripts/generate

```diff
@@ -108,6 +109,30 @@ def ('fromJust')
     ([$.Maybe (a), a])
     (maybe_ (() => { throw new Error ('fromJust applied to Nothing'); }) (I));
 
+//    replace :: RegExp -> String -> String -> String
+const replace =
+def ('replace')
+    ({})
+    ([$.RegExp, $.String, $.String, $.String])
+    (pattern => replacement => text => text.replace (pattern, replacement));
+
+//    replace_ :: RegExp -> (Array (Maybe String) -> String) -> String -> String
```

Comment:
Regarding naming, maybe:

```hs
replace :: RegExp -> (Array (Maybe String) -> String) -> String -> String
replace_ :: RegExp -> (Array String -> String) -> String -> String
replaceBy :: RegExp -> String -> String -> String
```

Reply:
Naming these functions is challenging because `'$1-$2-$3'` and `` ([$1, $2, $3]) => `${$1}-${$2}-${$3}` `` both specify how a replacement string is to be determined. Any sensible name for one function makes just as much sense for the other function.

Reply:
I moved this comment to https://github.com/sanctuary-js/sanctuary/pull/685#issuecomment-664196380, where I think it makes more sense.